osu.Game/Screens/Play/HUD/JudgementCounter/JudgementCountController.cs

```diff
@@ -53,8 +52,40 @@ protected override void LoadComplete()
         {
             base.LoadComplete();
 
+            scoreProcessor.OnResetFromReplayFrame += updateAllCounts;
             scoreProcessor.NewJudgement += judgement => updateCount(judgement, false);
             scoreProcessor.JudgementReverted += judgement => updateCount(judgement, true);
+
+            updateAllCounts();
+        }
+
+        private void updateAllCounts()
+        {
+            // This flow is made to handle cases of watching from the middle of a replay / spectating session.
+            //
+            // Once we get an initial state, we can rely on `NewJudgement` and `JudgementReverted`, so
+            // as a preemptive optimisation, only do a full re-sync if we have all-zero counts.
+            bool hasCounts = false;
+
+            foreach (var r in results)
+            {
+                if (r.Value.ResultCount.Value > 0)
+                {
+                    hasCounts = true;
+                    break;
+                }
+            }
```

Comment:
This breaks the PR, because it appears that the resets-from-replay-frame occur at the end of a frame bundle, at which point part of gameplay has been replayed and very likely incurred an actual judgement through the expected flows.

Reply:
I've applied a simple fix which looks good on paper and seems to work in initial testing. There's a brief period where counts won't be updated (until the first reset event) but I think this is okay.